tests/unittests/test_datasource/test_azure_helper.py

```diff
@@ -365,6 +459,113 @@ def test_parse_certificates(self, mock_decrypt_certs):
+        super(TestGoalStateHealthReporter, self).setUp()
+        patches = ExitStack()
+        self.addCleanup(patches.close)
+
+        patches.enter_context(
+            mock.patch.object(azure_helper.time, 'sleep', mock.MagicMock()))
+        self.read_file_or_url = patches.enter_context(
+            mock.patch.object(azure_helper.url_helper, 'read_file_or_url'))
+
+        self.post = patches.enter_context(
+            mock.patch.object(azure_helper.AzureEndpointHttpClient,
+                              'post'))
+
+        self.GoalState = patches.enter_context(
+            mock.patch.object(azure_helper, 'GoalState'))
+        self.GoalState.return_value.container_id = \
+            self.default_parameters['container_id']
+        self.GoalState.return_value.instance_id = \
+            self.default_parameters['instance_id']
+        self.GoalState.return_value.incarnation = \
+            self.default_parameters['incarnation']
+
+    def _get_formatted_health_report_xml_string(self, **kwargs):
+        return HEALTH_REPORT_XML_TEMPLATE.format(**kwargs)
+
+    def _get_report_ready_health_document(self):
+        return self._get_formatted_health_report_xml_string(
+            incarnation=self.default_parameters['incarnation'],
+            container_id=self.default_parameters['container_id'],
+            instance_id=self.default_parameters['instance_id'],
+            health_status=self.provisioning_success_status,
+            health_detail_subsection='')
+
+    def test_send_ready_signal_sends_post_request(self):
+        health_document = self._get_report_ready_health_document()
+        client = azure_helper.AzureEndpointHttpClient(mock.MagicMock())
+        reporter = azure_helper.GoalStateHealthReporter(
+            azure_helper.GoalState(mock.MagicMock(), mock.MagicMock()),
+            client, self.test_endpoint)
+        reporter.send_ready_signal()
+        self.assertEqual(1, self.post.call_count)
+        self.assertEqual(
+            mock.call(
+                self.test_url,
+                data=health_document,
+                extra_headers=self.test_default_headers),
+            self.post.call_args)
+
+    def test_send_ready_signal_health_document(self):
+        health_document = self._get_report_ready_health_document()
+        reporter = azure_helper.GoalStateHealthReporter(
+            azure_helper.GoalState(mock.MagicMock(), mock.MagicMock()),
+            azure_helper.AzureEndpointHttpClient(mock.MagicMock()),
+            self.test_endpoint)
+        generated_health_document = reporter.build_report(
+            incarnation=self.default_parameters['incarnation'],
+            container_id=self.default_parameters['container_id'],
+            instance_id=self.default_parameters['instance_id'],
+            status=self.provisioning_success_status)
+        self.assertEqual(health_document, generated_health_document)
+        self.assertIn(str(self.default_parameters['incarnation']),
```

Comment:
This is just testing that "1" appears in the generated document, somewhere, I think? The first line of the template includes "1" so this test will always pass, regardless of whether or not the incarnation value is used.

I suggest either using a large number that is unlikely to appear in the template in future or, probably better, asserting that the full XML element is present (i.e. test for `<GoalStateIncarnation>1</GoalStateIncarnation>`).

(The `container_id` and `instance_id` assertions don't have the same problem as they are using unique strings, but it would still be good to assert that the XML element is correct.)